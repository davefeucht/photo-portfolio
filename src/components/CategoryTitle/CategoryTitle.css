.category-title {
  width: 100%;
  height: min(1.7rem, 2.2vw);
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  bottom: 0
}

.category-title-text {
  color: #FFFFFF;
  /*
  writing-mode: vertical-rl;
  text-orientation: sideways;
  */
  margin-left: .3rem;
  font-size: min(1.5rem, 2vw);
}