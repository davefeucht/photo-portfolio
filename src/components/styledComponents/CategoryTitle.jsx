import styled from "styled-components";

const CategoryTitle = styled.div`
  font-size: 120%;
  font-weight: bold;
`;

export default CategoryTitle;