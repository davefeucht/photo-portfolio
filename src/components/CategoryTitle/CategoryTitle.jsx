/*****************
* CategoryTitle component displays the category title over a CategoryThumbnail for a particular category.
*****************/

import React from 'react';
import PropTypes from 'prop-types';
import './CategoryTitle.css';

const CategoryTitle = ({ name }) => {
  return (
    <div className="category-title">
      <div className="category-title-text">{name}</div>
    </div>
  )
};

CategoryTitle.displayName = 'CategoryTitle';

CategoryTitle.propTypes = {
  name: PropTypes.string.isRequired
}

export default CategoryTitle;