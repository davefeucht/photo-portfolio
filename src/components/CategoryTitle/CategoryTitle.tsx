/** ***************
* CategoryTitle component displays the category title over a CategoryThumbnail for a particular category.
**************** */

import './CategoryTitle.css';

import * as React from 'react';

interface CategoryTitleProps {
    title: string
}

const CategoryTitle: React.FC<CategoryTitleProps> = ({ title }) => {
    return (
        <div className="category-title">
            <div className="category-title-text">{title}</div>
        </div>
    );
};

CategoryTitle.displayName = 'CategoryTitle';

export default CategoryTitle;
