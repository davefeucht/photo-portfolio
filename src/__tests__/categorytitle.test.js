import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CategoryTitle from '../components/CategoryTitle/CategoryTitle';

const name = 'Test Category';

test('CategoryTitle displays', async () => {
    let component;
    let tree;
    component = renderer.create(
        <MemoryRouter>
            <CategoryTitle title={name} ></CategoryTitle>
        </MemoryRouter>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
