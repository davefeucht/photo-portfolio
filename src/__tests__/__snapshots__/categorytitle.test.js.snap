// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryThumbnail displays 1`] = `
<div
  className="category-title"
>
  <div
    className="category-title-text"
  >
    Test Category
  </div>
</div>
`;
